src/lib.rs

```diff
@@ -0,0 +1,54 @@
+#[macro_use]
+extern crate nickel;
+extern crate openssl;
+extern crate rand;
+extern crate rustc_serialize;
+extern crate byteorder;
+extern crate time;
+extern crate cookie;
+extern crate plugin;
+extern crate typemap;
+
+use std::fmt::Debug;
+use rustc_serialize::{Encodable, Decodable};
+use std::any::Any;
+use nickel::{Response, Middleware, MiddlewareResult, cookies, NickelError, Session, SessionStore};
+use nickel::status::StatusCode::Forbidden;
+
+pub type ValidateUserFunc<T> = Box<Fn(&T) -> bool + Send + Sync>;
+
+pub struct Authorizer<T, D> {
+    authorize: ValidateUserFunc<T>,
+    access_granted: Box<Middleware<D>+ Sized + Send + Sync>,
```

Comment:
Probably! I haven't read up on static and dynamic dispatch. Would the compiler be able to figure it out after what I just pushed?